vertx-db2-client/src/main/java/io/vertx/db2client/impl/drda/DRDAConnectResponse.java

```diff
@@ -1154,6 +1154,191 @@ private void parseACCSECreply(int securityMechanism) {
+    //     The application server rejected establishment of the connection.
+    //     SQLCODE : -30061
+    //     The database alias or database name <name> was not found at the remote node.
+    //     The statement cannot be processed.
+    //
+    //
+    // Returned from Server:
+    // SVRCOD - required  (8 - ERROR)
+    // RDBNAM - required
+    private void parseRDBNFNRM() {
+        boolean svrcodReceived = false;
+        int svrcod = CodePoint.SVRCOD_INFO;
+        boolean rdbnamReceived = false;
+        String rdbnam = null;
+
+        parseLengthAndMatchCodePoint(CodePoint.RDBNFNRM);
+        pushLengthOnCollectionStack();
+        int peekCP = peekCodePoint();
+
+        while (peekCP != END_OF_COLLECTION) {
+
+            boolean foundInPass = false;
+
+            if (peekCP == CodePoint.SVRCOD) {
+                foundInPass = true;
+                svrcodReceived = checkAndGetReceivedFlag(svrcodReceived);
+                svrcod = parseSVRCOD(CodePoint.SVRCOD_ERROR, CodePoint.SVRCOD_ERROR);
+                peekCP = peekCodePoint();
+            }
+
+            if (peekCP == CodePoint.RDBNAM) {
+                foundInPass = true;
+                rdbnamReceived = checkAndGetReceivedFlag(rdbnamReceived);
+                rdbnam = parseRDBNAM(true);
+                peekCP = peekCodePoint();
+            }
+            
+            if (peekCP == CodePoint.SRVDGN) {
+            	foundInPass = true;
+            	String serverDiagnostics = parseSRVDGN();
+            	// TODO: @AGG Log the server diagnostics here
+            	peekCP = peekCodePoint();
+            }
+
+            if (!foundInPass) {
+            	throwUnknownCodepoint(peekCP);
+            }
+
+        }
+        popCollectionStack();
+        if (!svrcodReceived)
+        	throwMissingRequiredCodepoint("SVRCOD", CodePoint.SVRCOD);
+        if (!rdbnamReceived)
+        	throwMissingRequiredCodepoint("RDBNAM", CodePoint.RDBNAM);
+
+//        netAgent_.setSvrcod(svrcod);
+        throw new DB2Exception("The requested database was not found: " + metadata.databaseName,
+        		SqlCode.RDB_NOT_FOUND, SQLState.NET_DATABASE_NOT_FOUND);
+//        agent_.accumulateChainBreakingReadExceptionAndThrow(new DisconnectException(agent_,
+//            new ClientMessageId(SQLState.NET_DATABASE_NOT_FOUND),
+//            netConnection.databaseName_));
+    }
+    
+    private void parseRdbAccessFailed() {
+        parseRDBAFLRM();
+
+        // an SQLCARD is returned if an RDBALFRM is returned.
+        // this SQLCARD always follows the RDBALFRM.
+        // TYPDEFNAM and TYPDEFOVR are MTLINC
+
+        if (peekCodePoint() == CodePoint.TYPDEFNAM) {
+            parseTYPDEFNAM();
+            parseTYPDEFOVR();
+        } else {
+            parseTYPDEFOVR();
+            parseTYPDEFNAM();
+        }
+
+        NetSqlca netSqlca = parseSQLCARD(null);
+        
+        //Check if the SQLCARD has null SQLException
+        if(netSqlca.getSqlErrmc() == null) {
```

Comment:
Are we guaranteed a non-null value for netSqlca before we call any methods?  I see in parseSQLCAGRP there is the possibility of a null value being returned.  Or is it ok if we throw a NPE at this point?

Reply:
A non-null SQLCARD should always be returned if a normal DB error occurs (which excludes situations where the DB just closes the connection). On this code path we're still under "normal" error path situations since we only get to this point if we have previously found and matched a `RDBAFLRM` message which must be followed by a SQLCARD